template_agent/src/routes/openai_chat.py

```diff
@@ -0,0 +1,328 @@
+"""OpenAI-compatible chat completions API endpoint.
+
+This module provides OpenAI API compatibility for chat completions,
+allowing existing OpenAI clients to interact with the template agent.
+"""
+
+import time
+import uuid
+from collections.abc import AsyncGenerator
+from typing import Any, Literal, cast
+
+from fastapi import APIRouter, HTTPException, Request
+from fastapi.responses import StreamingResponse
+
+from template_agent.src.core.exceptions.exceptions import AppException, AppExceptionCode
+from template_agent.src.core.manager import AgentManager
+from template_agent.src.schema import (
+    OpenAIChatRequest,
+    OpenAIChatResponse,
+    OpenAIChatStreamChoice,
+    OpenAIChatStreamDelta,
+    OpenAIChatStreamResponse,
+    OpenAIChoice,
+    OpenAIMessage,
+    OpenAIUsage,
+    StreamRequest,
+)
+from template_agent.src.settings import settings
+from template_agent.utils.pylogger import get_python_logger
+
+router = APIRouter()
+logger = get_python_logger(settings.PYTHON_LOG_LEVEL)
+
+
+def _app_exception_to_http_exception(app_exception: AppException) -> HTTPException:
+    """Convert AppException to HTTPException for FastAPI compatibility.
+
+    Args:
+        app_exception: The AppException to convert.
+
+    Returns:
+        HTTPException with appropriate status code and detail.
+    """
+    return HTTPException(
+        status_code=app_exception.response_code,
+        detail=f"{app_exception.detail_message} (Error: {app_exception.error_code})",
+    )
```

Comment:
@tuhinsharma121, This converted Http exception is going to get caught at the app level exception handler and will get converted back into an app exception and thrown to user. I may be missing the full context. Can you please share your thoughts? Thanks.